packages/firestore/test/unit/specs/bundle_spec.test.ts

```diff
@@ -270,13 +292,93 @@ describeSpec('Bundles:', ['no-ios', 'no-android'], () => {
     );
   });
 
+  specTest('Bundles query can be resumed from same query.', [], () => {
+    const query1 = query('collection');
+    const docA = doc('collection/a', 100, { key: 'a' });
+    const bundleString1 = bundleWithDocumentAndQuery(
+      {
+        key: docA.key,
+        readTime: 500,
+        createTime: 250,
+        updateTime: 500,
+        content: { value: 'b' }
+      },
+      { name: 'bundled-query', readTime: 400, query: query1 }
+    );
+
+    return spec()
+      .loadBundle(bundleString1)
+      .userListens(query1, 400)
+      .expectEvents(query1, {
+        added: [doc('collection/a', 500, { value: 'b' })],
+        fromCache: true
+      });
+  });
+
+  specTest(
+    'Bundles query can be loaded and resumed from different tabs',
+    ['multi-client'],
+    () => {
+      const query1 = query('collection');
+      const query2 = query('collection', filter('value', '==', 'c'));
+      const docA = doc('collection/a', 100, { value: 'a' });
+      const bundleString1 = bundleWithDocumentAndQuery(
+        {
+          key: docA.key,
+          readTime: 500,
+          createTime: 250,
+          updateTime: 500,
+          content: { value: 'b' }
+        },
+        { name: 'bundled-query', readTime: 400, query: query1 }
+      );
+
+      const bundleString2 = bundleWithDocumentAndQuery(
+        {
+          key: docA.key,
+          readTime: 600,
+          createTime: 250,
+          updateTime: 550,
+          content: { value: 'c' }
+        },
+        { name: 'bundled-query', readTime: 560, query: query2 }
+      );
+
+      return (
+        client(0)
+          .loadBundle(bundleString1)
+          // Read named query from loaded bundle by primary.
+          .client(1)
+          .userListens(query1, 400)
+          .expectEvents(query1, {
+            added: [doc('collection/a', 500, { value: 'b' })],
+            fromCache: true
+          })
+          // Loads a newer bundle.
+          .loadBundle(bundleString2)
+          .expectEvents(query1, {
+            modified: [doc('collection/a', 550, { value: 'c' })],
+            fromCache: true
+          })
+          .userUnlistens(query1)
```

Comment:
This test would be more interesting if you didn't unlisten here, since you then have to provide an `expectActiveTarget` with a read time for `client(0)`.